test/Files.test.js

```diff
@@ -12,9 +12,15 @@ const ujoinFiles = (dirs, fakeFiles) => {
   return fakeFiles.map(f => ujoin(...dirs, f))
 }
 const fakeFile = 'fake/file.txt'
+let initWithNewCredsMock = jest.spyOn(Files.prototype, '_initWithNewCreds')
 
 beforeEach(() => {
   expect.hasAssertions()
+  initWithNewCredsMock.mockReset()
+})
+
+afterAll(() => {
+  initWithNewCredsMock.mockRestore()
```

Comment:
I don't think you need to restore the mock , your spy is not replacing the method